app/src/ui/lib/props-type.ts

```diff
@@ -0,0 +1,6 @@
+// based on https://codewithstyle.info/TypeScript-conditional-types-real-life-example/
+export type PropsType<C> = C extends React.Component<infer P>
```

Comment:
This little helper function means we can avoid `export`-ing interfaces on props.

In the test below I was able to type-check the props I was setting up using this:

```tsx
let defaultProps: PropsType<PushPullButton>
```